azure/services/virtualmachines/spec.go

```diff
@@ -291,6 +293,48 @@ func (s *VMSpec) generateSecurityProfile() (*compute.SecurityProfile, error) {
 	}, nil
 }
 
+func (s *VMSpec) generateDiagnosticsProfile() (*compute.DiagnosticsProfile, error) {
+	if s.DiagnosticsProfile.Boot == nil {
+		return &compute.DiagnosticsProfile{
+			BootDiagnostics: &compute.BootDiagnostics{
+				Enabled: to.BoolPtr(true),
```

Comment:
we shouldn't be applying some defaults at runtime and others in webhooks, see my comment below for making Boot required when DiagnosticsProfile is specified or making sure we always default Boot

Reply:
👍